integration-tests/src/router.rs

```diff
@@ -1,623 +1,1325 @@
-		//Arrange
-		assert_trader_non_native_balance(BOB_INITIAL_AUSD_BALANCE, AUSD);
+			assert_trader_bsx_balance(BOB_INITIAL_BSX_BALANCE - amount_to_sell);
+			assert_trader_non_native_balance(amount_out, KSM);
 
-		let amount_to_sell = 10;
-		let limit = 0;
-		let trades = vec![Trade {
-			pool: PoolType::XYK,
-			asset_in: BSX,
-			asset_out: AUSD,
-		}];
-
-		//Act and Assert
-		assert_noop!(
-			Router::sell(
+			expect_basilisk_events(vec![pallet_route_executor::Event::RouteExecuted {
+				asset_in: BSX,
+				asset_out: KSM,
+				amount_in: amount_to_sell,
+				amount_out,
+			}
+			.into()]);
+		});
+	}
+
+	#[test]
+	fn sell_should_work_when_route_contains_multiple_trades() {
+		TestNet::reset();
+
+		Basilisk::execute_with(|| {
+			//Arrange
+			create_xyk_pool(BSX, AUSD);
+			create_xyk_pool(AUSD, MOVR);
+			create_xyk_pool(MOVR, KSM);
+
+			assert_trader_non_native_balance(BOB_INITIAL_AUSD_BALANCE, AUSD);
+			assert_trader_non_native_balance(0, MOVR);
+			assert_trader_non_native_balance(0, KSM);
+
+			let amount_to_sell = 10 * UNITS;
+			let limit = 0;
+			let trades = vec![
+				Trade {
+					pool: PoolType::XYK,
+					asset_in: BSX,
+					asset_out: AUSD,
+				},
+				Trade {
+					pool: PoolType::XYK,
+					asset_in: AUSD,
+					asset_out: MOVR,
+				},
+				Trade {
+					pool: PoolType::XYK,
+					asset_in: MOVR,
+					asset_out: KSM,
+				},
+			];
+
+			//Act
+			assert_ok!(Router::sell(
 				Origin::signed(TRADER.into()),
 				BSX,
-				AUSD,
-				amount_to_sell * UNITS,
+				KSM,
+				amount_to_sell,
 				limit,
 				trades
-			),
-			pallet_xyk::Error::<basilisk_runtime::Runtime>::TokenPoolNotFound
-		);
+			));
 
-		assert_trader_bsx_balance(BOB_INITIAL_BSX_BALANCE);
-		assert_trader_non_native_balance(BOB_INITIAL_AUSD_BALANCE, AUSD);
-	});
-}
+			//Assert
+			let amount_out = 1_054_553_059_484_u128;
+
+			assert_trader_bsx_balance(BOB_INITIAL_BSX_BALANCE - amount_to_sell);
+			assert_trader_non_native_balance(amount_out, KSM);
+			assert_trader_non_native_balance(BOB_INITIAL_AUSD_BALANCE, AUSD);
+			assert_trader_non_native_balance(0, MOVR);
+
+			expect_basilisk_events(vec![pallet_route_executor::Event::RouteExecuted {
+				asset_in: BSX,
+				asset_out: KSM,
+				amount_in: amount_to_sell,
+				amount_out,
+			}
+			.into()]);
+		});
+	}
+
+	#[test]
+	fn sell_should_fail_when_there_is_no_pool_for_specific_asset_pair() {
+		TestNet::reset();
+
+		Basilisk::execute_with(|| {
+			//Arrange
+			assert_trader_non_native_balance(BOB_INITIAL_AUSD_BALANCE, AUSD);
+
+			let amount_to_sell = 10;
+			let limit = 0;
+			let trades = vec![Trade {
+				pool: PoolType::XYK,
+				asset_in: BSX,
+				asset_out: AUSD,
+			}];
+
+			//Act and Assert
+			assert_noop!(
+				Router::sell(
+					Origin::signed(TRADER.into()),
+					BSX,
+					AUSD,
+					amount_to_sell * UNITS,
+					limit,
+					trades
+				),
+				pallet_xyk::Error::<basilisk_runtime::Runtime>::TokenPoolNotFound
+			);
+
+			assert_trader_bsx_balance(BOB_INITIAL_BSX_BALANCE);
```

Comment:
It's not necessary to test anything after `assert_noop`.

Reply:
I am doing these checks in the tests because in a route we can do multiple trades, and for each trade there is a currency transfer happening. 

If one of them fails I want to be sure that everything is rolled back.

I could introduce some helper methods to make these assertion intents more explicit, for exa,ple: `asset_that_trader_bas_balance_has_not_changed`